### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by marekkawalski**

### CODE REVIEW

File: frontend/smart-kickers-game/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "smart-kickers-game",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.4",
```
Comment (VOID404) [inline frontend/smart-kickers-game/package.json:6]: Shouldn't testing libraries should be in `dev-dependencies`?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module remote
```
Comment (VOID404) [inline go.mod:1]:
```suggestion
module github.com/HackYourCareer/SmartKickers
```

File: internal/echo/echo.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
+package echo
+
+import (
+	"encoding/json"
+	"github.com/gorilla/websocket"
+	"log"
+	"net/http"
+	"remote/pkg/messages"
```
Comment (VOID404) [inline internal/echo/echo.go:8]: `remote` -> `github.com/HackYourCareer/SmartKickers`

File: main.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"flag"
+	"log"
+	"net/http"
+	"remote/internal/echo"
+
+	"github.com/gorilla/mux"
+)
+
+var nodeAddr = flag.String("addr1", "0.0.0.0:3000", "nodeAddr")
```
Comment (VOID404) [inline main.go:12]: `nodeAddr` and `reactAddr` could be grouped:
```
var (
    nodeAddr = ...
    reactAddr = ...
)

File: main.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+var reactAddr = flag.String("addr2", "0.0.0.0:3006", "reactAddr")
+
+func main() {
+	flag.Parse()
+	log.SetFlags(0)
+
+	r1 := mux.NewRouter()
+	r1.HandleFunc("/", echo.Echo)
+	r1.HandleFunc("/shot", echo.ReceiveShotMsg)
+	go http.ListenAndServe(*nodeAddr, r1)
```
Comment (VOID404) [inline main.go:22]: Missing error handling

File: main.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+	r2 := mux.NewRouter()
+	r2.HandleFunc("/csc", echo.SendScoreHandler)
+	http.ListenAndServe(*reactAddr, r2)
```
Comment (VOID404) [inline main.go:26]: Missing error handling

File: main.go (hunk 5 above)
Comment (VOID404) [inline main.go:22]: is there really a need for 2 http servers?

File: main.go [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+	"github.com/gorilla/mux"
+)
+
+var nodeAddr = flag.String("addr1", "0.0.0.0:3000", "nodeAddr")
+var reactAddr = flag.String("addr2", "0.0.0.0:3006", "reactAddr")
+
+func main() {
+	flag.Parse()
+	log.SetFlags(0)
```
Comment (VOID404) [inline main.go:17]: I wouldn't remove all the useful info from logs

File: internal/echo/echo.go [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+		GameId: dm.TableId,
+	}
+
+	msg, err := json.Marshal(message)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	//	Send json back through connected websocket
+	c.WriteMessage(mt, msg)
```
Comment (VOID404) [inline internal/echo/echo.go:27]: Unchecked error

File: internal/echo/echo.go [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+		//	Receive message from connection, check for errors
+		mt, message, err := c.ReadMessage()
+		if err != nil {
+			log.Println("read:", err)
+			return
+		}
+
+		//	Resolve given json into provided struct
+		var dm messages.DispatcherMsg
+		err = json.Unmarshal([]byte(message), &dm)
```
Comment (VOID404) [inline internal/echo/echo.go:69]: Converting from `[]byte` to `[]byte`

File: internal/echo/shotmessage.go [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+	for {
+		_, message, err := c.ReadMessage()
+		if err != nil {
+			log.Println("read:", err)
+			break
+		}
+
+		var shotMsg messages.ShotMsg
+		_ = json.Unmarshal(message, &shotMsg)
```
Comment (VOID404) [inline internal/echo/shotmessage.go:24]: Error handling

File: internal/echo/shotmessage.go [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+		var shotParams messages.ShotParams
+		if len(shotMsg.Params) > 0 {
+			_ = json.Unmarshal(shotMsg.Params[0], &shotParams)
```
Comment (VOID404) [inline internal/echo/shotmessage.go:28]: Error handling

File: pkg/messages/resetscore_test.go [hunk 12]
```diff
@@ -0,0 +1,27 @@
+package messages
+
+import (
+	"testing"
+)
+
+func Test_resetScore(t *testing.T) {
```
Comment (VOID404) [inline pkg/messages/resetscore_test.go:7]: camelCase

File: internal/echo/echo_test.go [hunk 13]
```diff
@@ -0,0 +1,140 @@
+package echo
```
Comment (VOID404) [inline internal/echo/echo_test.go:1]: Consider putting tests in a separate package `echo_test`

File: pkg/messages/DispatcherMsg.go [hunk 14]
```diff
@@ -0,0 +1,12 @@
+package messages
+
+type DispatcherMsg struct {
+	MsgType   string  `json:"type,omitempty"`
+	Origin    string  `json:"origin,omitempty"`
+	TableId   string  `json:"id,omitempty"`
```
Comment (VOID404) [inline pkg/messages/DispatcherMsg.go:6]:
```suggestion
	TableID   string  `json:"id,omitempty"`
```

File: pkg/messages/DispatcherRes.go [hunk 15]
```diff
@@ -0,0 +1,21 @@
+package messages
+
+import "encoding/json"
+
+type DispatcherResponse struct {
+	GameId    string `json:"start,omitempty"`
```
Comment (VOID404) [inline pkg/messages/DispatcherRes.go:6]:
```suggestion
	GameID    string `json:"start,omitempty"`
```

File: pkg/messages/DispatcherRes.go [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+func ReturnResponse(dr *DispatcherResponse) []byte {
+	res, err := json.Marshal(DispatcherResponse{GameId: dr.GameId, GameEnded: dr.GameEnded})
+	if err != nil {
+		panic(err)
+	}
+	return res
+}
+
+func (dr *DispatcherResponse) New(gameId string, endId int) {
```
Comment (VOID404) [inline pkg/messages/DispatcherRes.go:18]:
```suggestion
func (dr *DispatcherResponse) New(gameID string, endId int) {
```

File: pkg/messages/resetscore_test.go [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+	var gameScore *GameScore = &GameScore{2, 0}
```
Comment (VOID404) [inline pkg/messages/resetscore_test.go:8]:
```suggestion
	gameScore := &GameScore{2, 0}
```

File: pkg/messages/DispatcherMsg.go [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+	X         float64 `json:"x,omitempty"`
+	Y         float64 `json:"y,omitempty"`
+	Timestamp string  `json:"timestamp,omitempty"`
+	Goal      int     `json:"goal,omitempty"`
+	Sequence  string  `json:"Sequence,omitempty"`
```
Comment (VOID404) [inline pkg/messages/DispatcherMsg.go:11]: All JSON fields start with a lowercase letter, except this one

File: internal/echo/echo.go [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+		if err != nil {
+			log.Println(err)
+		}
+
+		//	Establish a connection if initial message is received
+		if dm.MsgType == "INITIAL" {
```
Comment (VOID404) [inline internal/echo/echo.go:75]: Switch over `MsgType`?

File: internal/echo/echo.go [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+	c, err := upgrader.Upgrade(w, r, nil)
+
+	if err != nil {
+		log.Fatalln("upgrade:", err)
+	}
+
+	return c
+}
+
+func Echo(w http.ResponseWriter, r *http.Request) {
```
Comment (VOID404) [inline internal/echo/echo.go:54]: What does this function do? How is it related to "echo"?

**Review by VOID404 — CHANGES_REQUESTED** [review]

File: frontend/smart-kickers-game/src/App.js [hunk 21]
```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import "./App.css";
+import { Button } from "./components/button";
+
+function App() {
+  let [blueScore, setBlueScore] = useState(0);
+  let [whiteScore, setWhiteScore] = useState(0);
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/App.js:7]: `const`

File: frontend/smart-kickers-game/src/App.spec.js [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+    expect(screen.getByTestId("blue-team-score")).toHaveTextContent("Blue: 1");
+  });
+
+  it("should update score on white goal message", async () => {
+    const { container } = render(<App />);
+    await ws.connected;
+
+    ws.send(JSON.stringify({ type: "whiteGoal" }));
+
+    expect(screen.getByTestId("blue-team-score")).toHaveTextContent("White: 1");
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/App.spec.js:37]: Why `blue-team-score` has `white: 1`?

File: frontend/smart-kickers-game/src/components/button.js [hunk 23]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import "./button.css";
+
+const STYLES = ["btn--primary", "btn--outline"];
+const SIZES = ["btn--medium", "btn--large"];
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/button.js:5]: There aren't styles and sizes, there are classNames.

File: frontend/smart-kickers-game/src/components/button.js [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+  type,
+  onClick,
+  buttonStyle,
+  buttonSize,
+}) => {
+  const checkButtonStyle = STYLES.includes(buttonStyle)
+    ? buttonStyle
+    : STYLES[0];
+
+  const checkButtonSize = SIZES.includes(buttonSize) ? buttonSize : SIZES[0];
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/button.js:18]: If it should be a generic button just send prop `className`

File: frontend/smart-kickers-game/src/components/button.js [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+
+  return (
+    <button
+      className={`btn ${checkButtonStyle} ${checkButtonSize}`}
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/components/button.js:22]: className={`btn ${classNames}}`

File: frontend/smart-kickers-game/README.md [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+
+This section has moved here: [https://facebook.github.io/create-react-app/docs/advanced-configuration](https://facebook.github.io/create-react-app/docs/advanced-configuration)
+
+### Deployment
+
+This section has moved here: [https://facebook.github.io/create-react-app/docs/deployment](https://facebook.github.io/create-react-app/docs/deployment)
+
+### `npm run build` fails to minify
+
+This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/README.md:70]: It's unnecessary, just make one good README or remove it.

File: frontend/smart-kickers-game/public/robots.txt [hunk 27]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
+Disallow:
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/public/robots.txt:3]: Just remove that.

File: frontend/smart-kickers-game/src/App.js [hunk 28]
```diff
@@ -0,0 +1,38 @@
...
+    };
+  });
+
+  return (
+    <>
+      <h1>Smart Kickers</h1>
+      <div className="game-result-container" data-testid="blue-team-score">
+        Blue: {blueScore}
+        {"   "}
+        White: {whiteScore}
```
Comment (mrCherry97) [inline frontend/smart-kickers-game/src/App.js:27]: It should be in separated `<span>` or `<p>` or something like this.

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

File: pkg/messages/DispatcherMsg.go (hunk 18 above)
Comment (Pichi00) [reply pkg/messages/DispatcherMsg.go:11]: It is a problem on a server side. We receive messages in this form.

**Pull request closed without merging**